[community] Load Models from Sources like `Civitai` into Existing Pipelines (#9986)
huggingface/diffusers: examples/model_search/README.md

> # Searching Models on Civitai and Hugging Face
> Please refer to the original library [here](https://pypi.org/project/auto-diffusers/)

Reviewer: ```suggestion
# Search models on Civitai and Hugging Face

The [auto_diffusers](https://github.com/suzukimain/auto_diffusers) library provides additional functionalities to Diffusers such as searching for models on Civitai and the Hugging Face Hub.
Please refer to the original library [here](https://pypi.org/project/auto-diffusers/)
```

> ## Installing the dependencies
> 
> Before running the scripts, make sure to install the library's training dependencies:
> 
> **Important**
> 
> To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the install up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment:
> ```bash
> git clone https://github.com/huggingface/diffusers
> cd diffusers
> pip install .
> ```
> 
> Then cd in the example folder and run:
> ```bash
> pip install -r requirements.txt
> ```
> 
> 

Reviewer: ```suggestion
## Installation

Before running the scripts, make sure to install the library's training dependencies:

> [!IMPORTANT]
> To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the installation up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment.

```bash
git clone https://github.com/huggingface/diffusers
cd diffusers
pip install .
```

Navigate to the example folder and install the `requirements.txt` file.

```bash
pip install -r requirements.txt
```

```

> ##  Example<a name = "Example"></a>

Reviewer: ```suggestion
##  Search for models
```

```bash
!wget https://raw.githubusercontent.com/suzukimain/diffusers/refs/heads/ModelSearch/examples/model_search/search_for_civitai_and_HF.py
```

```python
# Search for Civitai

from search_for_civitai_and_HF import CivitaiSearchPipeline
from diffusers import StableDiffusionPipeline

model_path = CivitaiSearchPipeline.for_civitai(

>     "any",

Reviewer: Would be nice to explicitly specify that `any` refers to `search_word`

    base_model="SD 1.5",
    download=True
)
pipe = StableDiffusionPipeline.from_single_file(model_path).to("cuda")

```

```python
# Search for Hugging Face

from search_for_civitai_and_HF import HFSearchPipeline
from diffusers import StableDiffusionPipeline

model_path = HFSearchPipeline.for_HF(
           "stable",
           checkpoint_format="diffusers",
           download = False
           )

pipe = StableDiffusionPipeline.from_pretrained(model_path).to("cuda")

# or

model_path = HFSearchPipeline.for_HF(
           "stable",
           checkpoint_format="single_file",
           download = False
           )

pipe = StableDIffusionPipeline.from_single_file(model_path).to("cuda")
```

> <a id="Details"></a>
> <details close>

Reviewer: ```suggestion
```

# Search models on Civitai and Hugging Face

The [auto_diffusers](https://github.com/suzukimain/auto_diffusers) library provides additional functionalities to Diffusers such as searching for models on Civitai and the Hugging Face Hub.
Please refer to the original library [here](https://pypi.org/project/auto-diffusers/)

## Installation

Before running the scripts, make sure to install the library's training dependencies:

> [!IMPORTANT]
> To make sure you can successfully run the latest versions of the example scripts, we highly recommend **installing from source** and keeping the installation up to date as we update the example scripts frequently and install some example-specific requirements. To do this, execute the following steps in a new virtual environment.

```bash
git clone https://github.com/huggingface/diffusers
cd diffusers

> pip install .

Reviewer: Need to close the code snippet here

```bash
!wget https://raw.githubusercontent.com/suzukimain/auto_diffusers/refs/heads/master/src/auto_diffusers/pipeline_easy.py
```

> ### Search for Civitai

Reviewer: ```suggestion
## Load from Civitai
```

```python
from pipeline_easy import (
    EasyPipelineForText2Image,
    EasyPipelineForImage2Image,
    EasyPipelineForInpainting,
)

# Text-to-Image
pipeline = EasyPipelineForText2Image.from_civitai(
    "search_word",
    base_model="SD 1.5",
).to("cuda")

# Image-to-Image
pipeline = EasyPipelineForImage2Image.from_civitai(
    "search_word",
    base_model="SD 1.5",
).to("cuda")

# Inpainting
pipeline = EasyPipelineForInpainting.from_civitai(
    "search_word",
    base_model="SD 1.5",
).to("cuda")
```

> ### Search for Hugging Face

Reviewer: ```suggestion
## Load from Hugging Face
```

```python
from pipeline_easy import (
    EasyPipelineForText2Image,
    EasyPipelineForImage2Image,
    EasyPipelineForInpainting,
)

# Text-to-Image
pipeline = EasyPipelineForText2Image.from_huggingface(
    "search_word",
    checkpoint_format="diffusers",
).to("cuda")

# Image-to-Image
pipeline = EasyPipelineForImage2Image.from_huggingface(
    "search_word",
    checkpoint_format="diffusers",
).to("cuda")

# Inpainting
pipeline = EasyPipelineForInpainting.from_huggingface(
    "search_word",
    checkpoint_format="diffusers",
).to("cuda")
```

> ### Application Examples

Reviewer: ```suggestion
## Search Civitai and Hugging Face
```

> ### Application Examples

Reviewer: ```suggestion
## Search Civitai and Hugging Face
```

|:---------------:|:----------------------:|:-------------:|:-----------------------------------------------------------------------------------:|
| search_word     | string, Path           | ー            | The search query string. Can be a keyword, Civitai URL, local directory or file path. |
| model_type      | string                 | `Checkpoint`  | The type of model to search for.  <br>(for example `Checkpoint`, `TextualInversion`, `Controlnet`, `LORA`, `Hypernetwork`, `AestheticGradient`, `Poses`)      |
| base_model      | string                 | None          | Trained model tag (for example  `SD 1.5`, `SD 3.5`, `SDXL 1.0`) |
| torch_dtype     | string, torch.dtype    | None          | Override the default `torch.dtype` and load the model with another dtype.     |
| force_download  | bool                   | False         | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
| cache_dir       | string, Path | None    | Path to the folder where cached files are stored. |
| resume          | bool   | False         | Whether to resume an incomplete download. |

> | token           | string | None          | API token for Civitai authentication. |

Reviewer: You can move this to the `## Search Civitai` section

Author: Corrected.

> ### `search_civitai` parameters

Reviewer: Move to `## Search Civitai and Hugging Face`

|:---------------------:|:-------------------:|:--------------:|:----------------------------------------------------------------:|
| search_word           | string, Path        | ー             | The search query string. Can be a keyword, Hugging Face URL, local directory or file path, or a Hugging Face path (`<creator>/<repo>`). |
| checkpoint_format     | string              | `single_file`  | The format of the model checkpoint.<br>● `single_file` to search for `single file checkpoint` <br>●`diffusers` to search for `multifolder diffusers format checkpoint` |
| torch_dtype           | string, torch.dtype | None           | Override the default `torch.dtype` and load the model with another dtype. |
| force_download        | bool                | False          | Whether or not to force the (re-)download of the model weights and configuration files, overriding the cached versions if they exist. |
| cache_dir             | string, Path        | None           | Path to a directory where a downloaded pretrained model configuration is cached if the standard cache is not used.   |

> | token                 | string, bool        | None           | The token to use as HTTP bearer authorization for remote files.  |

Reviewer: Move this to the `## Search Hugging Face` section

> ### `search_huggingface` parameters

Reviewer: Move to `## Search Civitai and Hugging Face` section
